quickfixj-core/src/main/java/quickfix/Session.java

```diff
@@ -2590,6 +2600,7 @@ private boolean sendRaw(Message message, int num) {
                     }
                 }
 
+                persist(message, num);
                 messageString = message.toString();
```

Comment:
Could you please swap these lines and pass the `messageString` into `persist()`. That avoids two calls to `toString()` which should be avoided since it recalculates some fields of the message.

Reply:
Sure, didn't realize it was that inefficient, sorry :).